kernel/kernel-api/src/main/java/io/delta/kernel/ScanBuilder.java

```diff
@@ -65,4 +68,17 @@ public interface ScanBuilder {
 
   /** @return Build the {@link Scan instance} */
   Scan build();
+
+  /**
+   * Build a Paginated Scan with optional page token and optional log replay hashsets.
+   *
+   * @param pageSize Maximum number of AddFiles to return in this page.
+   * @param pageToken Optional page token representing the current scan position; null to start from
+   *     the beginning.
+   * @param logReplayHashsets Optional tombstone hashsets to filter out deleted files; null if not
+   *     applicable.
+   * @return A {@link PaginatedScan} configured for pagination.
+   */
+  PaginatedScan buildPaginated(
+      long pageSize, @Nullable Row pageToken, @Nullable ColumnarBatch logReplayHashsets);
```

Comment:
let's not use `@Nullable`

let's ignore `logReplayHashsets` for now entirely

if the page token is optional, let's clearly indicate that using java `Optional`